bst.rb

```diff
@@ -224,99 +240,260 @@ def rh(value, node = self.root)
+    end
+
+    if option == 5
+      ll.printll()
+      next
+    end
+  end
+end
+
+def bst_menu(t)
+
+  loop do
+    puts "Select from the given options"
+    puts "  Enter 0: to insert multiple elements"
+    puts "  Enter 1: to insert value"
+    puts "  Enter 2: to delete value"
+    puts "  Enter 3: to print BST"
+    puts "  Enter 4: to find Max"
+    puts "  Enter 5: to find Min"
+    puts "  Enter 6: to find if value exists"
+    puts "  Enter 7: to print all paths from root"
+    puts "  Enter 8: to load the last saved bst"
+    puts "  Enter quit to save and exit to main menu"
+    puts "  Enter anything else to go to main menu"
+
+    option = gets.chomp
+    system("clear")
+    if !option.is_i?
+      t.save() if option.downcase == "quit"
+      break
+    end
+    option = option.to_i
+
+    if option == 0
+      puts "enter multiple elements sepreated with ,"
+      arr = gets.chomp.split(",").map(&:to_i)
+      arr.each{|i| t.insert(i)}
+      puts "\n\n"
+      next
+    end
+
+    if option == 1
+      puts "Enter the value you want to insert"
+      v = gets.to_i
+      t.insert(v)
+      puts "\n#{v} was inserted in the tree"
+      puts "size of the tree now is #{t.size} \n\n"
+      next
+    end
+
+    if option == 2
+      puts "Enter the value you want to delete"
+      v = gets.to_i
+      if t.does_it_contains(v)
+        t.remove(v)
+        puts "\n#{v} was deleted from the tree"
+        puts "size of the tree now is #{t.size} \n\n"
+      else
+        puts "value not found can't delete it \n\n"
+      end
+      next
+    end
+
+    if option == 3
+      puts "Printing the tree in InOrder"
+      t.print_inorder()
+      puts "end\n Printing the tree in PreOrder"
+      t.print_preorder()
+      puts "end\n Printing the tree in PostOrder"
+      t.print_postorder()
+      puts "end\n Printing the tree in LevelOrder"
+      t.print_levelorder()
+      puts "end\n\n"
+      next
+    end
+
+    if option == 4
+      puts "the maximum is #{t.find_max().value}\n\n"
+      next
+    end
+
+    if option == 5
+      puts "the minimum is #{t.find_min().value}\n\n"
+      next
+    end
+
+    if option == 6
+      puts "Enter value to search it in treee"
+      v = t.does_it_contains(gets.to_i)
+      puts v ? "value exists" : "value does't exists"
+      next
+    end
+
+    if option == 7
+      puts "Printing all paths from Root"
+      t.print_paths()
+      puts "\n\n"
+      next
+    end
+
+    if option == 8
+      t.load_data()
+      puts "Data loaded from the file\n\n"
+    end
+  end
+end
+
+loop do
+  # option to choose between LL and bst
+  puts "Select the tool to use"
+  puts "  Enter 1: for using BST functions"
+  puts "  Enter 2: for using LL functions"
+  puts "  Enter 'exit' to quit the tool"
+  o = gets.chomp
+  break if o.downcase == "exit"
+  o = o.to_i
+  system("clear")
+  bst_menu(t) if o == 1
+  ll_menu(ll) if o == 2
+  puts "Incorrect input try again\n" if o!=1 && o!=2
 end
+system("clear")
```

Comment:
no need of braces for no parameter functions
no need to add braces while calling no parameter functions
if node can replace if node != null , as if node is super set of these commands